pkg/kv/kvnemesis/kvnemesis_test.go

```diff
@@ -139,17 +209,45 @@ type kvnemesisTestCfg struct {
 	numSteps     int
 	concurrency  int
 	seedOverride int64
+	// The two knobs below inject illegal lease index errors and, for the
+	// resulting reproposals, reproposal errors. The injection is stateful and
+	// remembers the keys on which the commands operated, and, per key, the
+	// probability is scaled down linearly based on the number of times we've
+	// injected an error. In other words, this can be set to 1.0 and some amount
+	// of progress would still be made.
+	//
+	// NB: to at least directionally preserve determinism, the rand for each dice
+	// roll is seeded from the uint64 represented by the key, so this shouldn't be
+	// considered truly random, but is random enough for the desired purpose.
+	invalidLeaseAppliedIndexProb float64 // [0,1)
+	injectReproposalErrorProb    float64 // [0,1)
 }
 
 func TestKVNemesisSingleNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
 	testKVNemesisImpl(t, kvnemesisTestCfg{
-		numNodes:     1,
-		numSteps:     defaultNumSteps,
-		concurrency:  5,
-		seedOverride: 0,
+		numNodes:                     1,
+		numSteps:                     defaultNumSteps,
+		concurrency:                  5,
+		seedOverride:                 0,
+		invalidLeaseAppliedIndexProb: 0.1,
+		injectReproposalErrorProb:    0.1,
+	})
+}
+
+func TestKVNemesisSingleNode_ReproposalChaos(t *testing.T) {
```

Comment:
I take it we don't really need any long-running nightlies for this, it's sufficient with a few runs as part of regular stress runs?